spec/workspace-spec.js

```diff
@@ -0,0 +1,2072 @@
+'use strict'
+
+/* global advanceClock, HTMLElement, waits */
+
+const path = require('path')
+const temp = require('temp').track()
+const TextEditor = require('../src/text-editor')
+const Workspace = require('../src/workspace')
+const Project = require('../src/project')
+const platform = require('./spec-helper-platform')
+const _ = require('underscore-plus')
+const fstream = require('fstream')
+const fs = require('fs-plus')
+
+describe('Workspace', () => {
+  let workspace
+  let setDocumentEdited
+
+  beforeEach(() => {
+    workspace = atom.workspace
+    workspace.resetFontSize()
+    spyOn(atom.applicationDelegate, 'confirm')
+    setDocumentEdited = spyOn(atom.applicationDelegate, 'setWindowDocumentEdited')
+    atom.project.setPaths([atom.project.getDirectories()[0].resolve('dir')])
+    waits(1)
+  })
+
+  afterEach(() => temp.cleanupSync())
+
+  describe('serialization', () => {
+    const simulateReload = () => {
+      const workspaceState = atom.workspace.serialize()
+      const projectState = atom.project.serialize({isUnloading: true})
+      atom.workspace.destroy()
+      atom.project.destroy()
+      atom.project = new Project({
+        notificationManager: atom.notifications,
+        packageManager: atom.packages,
+        confirm: atom.confirm.bind(atom),
+        applicationDelegate: atom.applicationDelegate
+      })
+      atom.project.deserialize(projectState)
+      atom.workspace = new Workspace({
+        config: atom.config,
+        project: atom.project,
+        packageManager: atom.packages,
+        grammarRegistry: atom.grammars,
+        deserializerManager: atom.deserializers,
+        notificationManager: atom.notifications,
+        applicationDelegate: atom.applicationDelegate,
+        viewRegistry: atom.views,
+        assert: atom.assert.bind(atom),
+        textEditorRegistry: atom.textEditors
+      })
+      return atom.workspace.deserialize(workspaceState, atom.deserializers)
+    }
+
+    describe('when the workspace contains text editors', () =>
+      it('constructs the view with the same panes', () => {
+        const pane1 = atom.workspace.getActivePane()
+        const pane2 = pane1.splitRight({copyActiveItem: true})
+        const pane3 = pane2.splitRight({copyActiveItem: true})
+        let pane4 = null
+
+        waitsForPromise(() => atom.workspace.open(null).then(editor => editor.setText('An untitled editor.')))
+
+        waitsForPromise(() =>
+          atom.workspace.open('b').then(editor => pane2.activateItem(editor.copy()))
+        )
+
+        waitsForPromise(() =>
+          atom.workspace.open('../sample.js').then(editor => pane3.activateItem(editor))
+        )
+
+        runs(() => {
+          pane3.activeItem.setCursorScreenPosition([2, 4])
+          pane4 = pane2.splitDown()
+        })
+
+        waitsForPromise(() =>
+          atom.workspace.open('../sample.txt').then(editor => pane4.activateItem(editor))
+        )
```

Comment:
Are you interested in switching this test to use `async` and `await`? If you feel like moving on, no pressure, but it can be kind of satisfying to :fire: all this `waitsForPromise` and `runs`.